cmd/dep/glide_config_test.go

```diff
@@ -0,0 +1,282 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bytes"
+	"log"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/golang/dep"
+	"github.com/golang/dep/internal/gps"
+)
+
+func TestGlideConvertProject(t *testing.T) {
+	loggers := &dep.Loggers{
+		Out:     log.New(os.Stdout, "", 0),
+		Err:     log.New(os.Stderr, "", 0),
+		Verbose: true,
+	}
+
+	f := glideFiles{
+		loggers: loggers,
+		yaml: glideYaml{
+			Imports: []glidePackage{
+				{
+					Name:       "github.com/sdboyer/deptest",
+					Repository: "https://github.com/fork/deptest.git",
+					Reference:  "v1.0.0",
+				},
+			},
+		},
+		lock: &glideLock{
+			Imports: []glidePackage{
+				{
+					Name:      "github.com/sdboyer/deptest",
+					Reference: "6a741be0cc55ecbe4f45690ebfd606a956d5f14a",
+				},
+			},
+		},
+	}
+
+	manifest, lock, err := f.convert("", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	d, ok := manifest.Dependencies["github.com/sdboyer/deptest"]
+	if !ok {
+		t.Fatal("Expected the manifest to have a dependency for 'github.com/sdboyer/deptest' but got none")
+	}
+
+	v := d.Constraint.String()
+	if v != "v1.0.0" {
+		t.Fatalf("Expected manifest constraint to be master, got %s", v)
+	}
+
+	if d.Source != "https://github.com/fork/deptest.git" {
+		t.Fatalf("Expected manifest source to be 'https://github.com/fork/deptest.git', got %s", d.Source)
+	}
+
+	if len(lock.P) != 1 {
+		t.Fatalf("Expected the lock to contain 1 project but got %d", len(lock.P))
+	}
+
+	p := lock.P[0]
+	if p.Ident().ProjectRoot != "github.com/sdboyer/deptest" {
+		t.Fatalf("Expected the lock to have a project for 'github.com/sdboyer/deptest' but got '%s'", p.Ident().ProjectRoot)
+	}
+
+	if p.Ident().Source != "https://github.com/fork/deptest.git" {
+		t.Fatalf("Expected locked source to be 'https://github.com/fork/deptest.git', got '%s'", p.Ident().Source)
+	}
+
+	lv := p.Version()
+	lpv, ok := lv.(gps.PairedVersion)
+	if !ok {
+		t.Fatalf("Expected locked version to be a PairedVersion but got %T", lv)
+	}
+
+	rev := lpv.Underlying()
+	if rev != "6a741be0cc55ecbe4f45690ebfd606a956d5f14a" {
+		t.Fatalf("Expected locked revision to be '6a741be0cc55ecbe4f45690ebfd606a956d5f14a', got %s", lv)
```

Comment:
Change `lv` to `rev`? `lv` results in the version string, in this case `v1.0.0`.